federated_learning/run_toxicity.py

```diff
@@ -146,45 +81,62 @@ def repeat_dataset(client_dataset, batch_size, epochs):
-@click.option('--epochs', default=10, help='the number of training epochs in client traning.')
-@click.option('--batchsize', default=32, help='the number of batch size.')
+@click.option('--epochs', default=500, help='the number of training epochs in client traning.')
+@click.option('--dataset_name', type=click.Choice(['Benchmark', 'NTP_PubChem_Bench.20201106', 'Ames_S9_minus.20201106']),
+              default='NTP_PubChem_Bench.20201106', help='set dataset name')
+@click.option('--batch_size', default=32, help='the number of batch size.')
+@click.option('--load_params', is_flag=True, default=False,
+              help='If set True, parameters are loaded and args below are ingored. To load params, model tuning should be executed previously.')
 @click.option('--lr', default=0.2, help='learning rate for the central model.')
-@click.option('--clientlr', default=0.001, help='learning rate for client models.')
-@click.option('--model', default='gcn', type=click.Choice(['gcn', 'gcn_super_node', 'gin', 'gin_super_node', 'gat', 'diff_pool']), help='support gcn, gin, gat.')
+@click.option('--clientlr', default=0.001, help='learning rate for client models. ignored when mode is normal')
+@click.option('--gnn_type', default='gcn', type=click.Choice(['gcn', 'gin']), help='support gcn, gin.')
 @click.option('--ratio', default=None, help='set ratio of the biggest dataset in total datasize.' +
               ' Other datasets are equally divided. (0, 1)')
-@click.option('--dataset_name', type=click.Choice(['Benchmark', 'NTP_PubChem_Bench.20201106', 'Ames_S9_minus.20201106']),
-              default='NTP_PubChem_Bench.20201106', help='set dataset name')
-def main(federated, rounds, clients, epochs, batchsize, lr, clientlr, model, ratio, dataset_name):
+def main(mode, rounds, clients, epochs, load_params, batch_size, lr, clientlr, gnn_type, ratio, dataset_name):
     logger = get_logger()
     subsets = clients + 2
-    logger.debug(f'federated = {federated}')
+    logger.debug(f'mode = {mode}')
     logger.debug(f'rounds = {rounds}')
     logger.debug(f'clients = {clients}')
     logger.debug(f'subsets = {subsets}')
     logger.debug(f'epochs = {epochs}')
-    logger.debug(f'batchsize = {batchsize}')
+    logger.debug(f'batch_size = {batch_size}')
     logger.debug(f'lr = {lr}')
     logger.debug(f'clientlr = {clientlr}')
-    logger.debug(f'model = {model}')
+    logger.debug(f'gnn_type = {gnn_type}')
     logger.debug(f'ratio = {ratio}')
     logger.debug(f'dataset_name = {dataset_name}')
 
-    if federated:
-        federated_learning(rounds, clients, epochs, batchsize,
-                           lr, clientlr, model, ratio, dataset_name)
+    if mode == "tuning":
+        model_tuning(dataset_name, epochs)
+    elif mode == "federated":
+        if load_params:
+            # NOTE: load params tunued without federated learning. `lr` is used as `clientlr`
+            params = load_best_params(dataset_name)
```

Comment:
I have implemented the code to allow tuning/loading parameters for each dataset. But the experiment was done using the model tuned with the "NTP_PubChem_Bench" dataset. So the code is slightly different from the experiment (argument of `load_best_params` was 'NTP_PubChem_Bench.20201106' in the experiment).